### PR TITLE
Added testing for error messages #92

### PR DESCRIPTION
**Pull request by Kalekdan**
---
Name: Tests for error messages
About: Redirected all calls to System.out.err through a new function which will store the latest error message. This allows for testing that a specific error message was thrown
---

**Is your PR related to a feature request or Bug report?**
If applicable, please list feature request number or bug report ID.
> #92 

**Describe your PR**
A clear and concise description of what your pull request is changing or adding.
> Adds testing for error messages to assert they are thrown as expected

**Describe intended use**
A clear and concise description of the intended use of the feature. Along with example code.
> Whenever displaying an error message using ```System.err.println(errMsg)```, instead use ```Mani.printAndStoreError(errMsg)```. This will still print the error message but will also store the latest message so it can be verified by tests.

**Is it in the form of a library**
 - [x] No

### CODE REVIEW

File: src/com/mani/lang/Modules/std/std.java [hunk 1]
```diff
@@ -117,6 +118,7 @@ public int arity() {
             public Object call(Interpreter interpreter, List<Object> arguments) {
                 interpreter.wipeMemory();
                 Std.loaded.clear();
+                Mani.hadRuntimeError = false;
```
Comment (Kalekdan) [inline src/com/mani/lang/Modules/std/std.java]: Made it so hat wiping memory also wipes the runtime error flag. This is to make sure the tests after those which are expected to error do not cause following tests to error out.

File: src/com/mani/lang/Modules/system/system.java [hunk 2]
```diff
@@ -68,6 +68,18 @@ public Object call(Interpreter interpreter, List<Object> arguments) {
                 return Mani.hadRuntimeError;
             }
         });
+
+        interpreter.addSTD("latestError", new ManiCallable() {
```
Comment (Kalekdan) [inline src/com/mani/lang/Modules/system/system.java]: Adds a new function to the "system" API to get the latest error message thrown.

Comment (crazywolf132) [reply src/com/mani/lang/Modules/system/system.java]: Awesome, i like the idea.

File: src/com/mani/lang/Modules/std/std.java (hunk 1 above)
Comment (crazywolf132) [reply src/com/mani/lang/Modules/std/std.java]: Good thinking. I would have overlooked this.

**Comment by crazywolf132** [comment]
Looks good to me. Just one thing though.

Maybe we make another Exception class for general errors, or we leverage the already existing one? Maybe this could also be a way to stop using `System.err.println`? I don't know if it would work here or not. Just an idea.

Merge when ready.

**Pull request merged**